common/persistence/data_interfaces.go

```diff
@@ -1204,6 +1205,52 @@ type (
 		SaveClusterMetadata(ctx context.Context, request *SaveClusterMetadataRequest) (bool, error)
 		DeleteClusterMetadata(ctx context.Context, request *DeleteClusterMetadataRequest) error
 	}
+
+	// HistoryTaskQueueManager is responsible for managing a queue of internal history tasks. It is currently unused,
+	// but we plan on using this to implement a DLQ for history tasks. This is called a history task queue manager, but
+	// the actual history task queues are not managed by this object. Instead, this object is responsible for managing
+	// a generic queue of history tasks (which is what the history task DLQ will be).
+	HistoryTaskQueueManager struct {
+		queue            QueueV2
+		serializer       *serialization.TaskSerializer
+		numHistoryShards int
+	}
+
+	// QueueKey identifies a history task queue. It is converted to a queue name using the GetQueueName method.
+	QueueKey struct {
```

Comment:
hmmm, ideally I think this should be an application layer concept and persistence package should depend on the application layer. But in this repo, the relationship is reversed. Guess we have to live with it.